src/App.tsx

```diff
@@ -1,56 +1,87 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { Person } from './types/Person';
+import { SearchField } from './components/SearchField';
+
+const findPersonBySlug = (personSlug: string): Person | null => {
+  return peopleFromServer.find(person => person.slug === personSlug) || null;
+};
+
+const getPeopleByQuery = (people: Person[], query: string): Person[] => people
+  .filter(person => {
+    const personName = person.name.toLowerCase();
+
+    return personName.includes(query.toLowerCase());
+  });
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedPersonSlug, setselectedPersonSlug] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  let selectedPerson = findPersonBySlug(selectedPersonSlug) || null;
+
+  const onChange = (value: string) => {
+    setSearchQuery(value);
+    if (selectedPerson
+        && selectedPerson.name.toLowerCase() !== value.toLowerCase()) {
+      setselectedPersonSlug('');
+      selectedPerson = null;
+    }
+  };
+
+  const onApplyChange = (value: string) => {
+    if (value !== appliedQuery) {
+      setAppliedQuery(value);
+    }
+  };
+
+  const visiblePersons = useMemo(() => {
+    if (appliedQuery) {
+      return getPeopleByQuery(peopleFromServer, searchQuery);
+    }
+
+    return peopleFromServer;
```

Comment:
ternary works better here
```suggestion
    return appliedQuery
      ? getPeopleByQuery(peopleFromServer, searchQuery)
      : peopleFromServer;
```